packages/image_picker/image_picker_for_web/lib/image_picker_for_web.dart

```diff
@@ -37,9 +37,63 @@ class ImagePickerPlugin extends ImagePickerPlatform {
     double maxHeight,
     int imageQuality,
     CameraDevice preferredCameraDevice = CameraDevice.rear,
-  }) {
+  }) async {
     String capture = computeCaptureAttribute(source, preferredCameraDevice);
-    return pickFile(accept: _kAcceptImageMimeType, capture: capture);
+    return pickFile(
+      accept: _kAcceptImageMimeType,
+      capture: await _resizeImage(
+        capture,
+        maxWidth.toInt(),
+        maxHeight.toInt(),
+        imageQuality,
+      ),
+    );
+  }
+
+  static Future<String> _resizeImage(
```

Comment:
I think this should return a `Future<PickedFile>`, with a PickedFile that contains the resized image. You can create a PickedFile from its bytes.